frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,244 @@ pub struct PoolRoles<AccountId> {
+					// before being converted back into a Perbill.
+					let max_allowed_increase =
+						Perbill::from_percent((t.max_increase * 100_u32) * intervals_passed);
+
+					attempted_increase > max_allowed_increase
+				},
+			) {
+				return true
+			}
+		}
+		false
+	}
+
+	/// Set the pool's commission.
+	///
+	/// Update commission based on `current`. If a `None` is supplied, allow the commission to be
+	/// removed without any change rate restrictions. If `change_rate` is present, update
+	/// `throttle_from` to the current block. If the supplied commission is zero, `None` will be
+	/// inserted and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		current: &Option<(Perbill, T::AccountId)>,
+	) -> DispatchResult {
+		if current.is_none() {
+			self.current = None;
+		} else {
+			let (commission, payee) = current.as_ref().ok_or(Error::<T>::NoCommissionSet)?;
+			ensure!(!self.throttling(&commission), Error::<T>::CommissionChangeThrottled);
+			ensure!(
+				self.max.map_or(true, |m| commission <= &m),
+				Error::<T>::CommissionExceedsMaximum
+			);
+
+			self.current =
+				if commission == &Zero::zero() { None } else { Some((*commission, payee.clone())) };
+		}
+
+		let _ = self.register_update();
+		Ok(())
+	}
+
+	/// Set the pool's maximum commission.
+	///
+	/// The pool's maximum commission can initially be set to any value, and only smaller values
+	/// thereafter. If larger values are attempted, this function will return a dispatch error.
+	///
+	/// If `current.0` is larger than the updated max commission value, `current.0` will also be
+	/// updated to the new maximum. This will also register a `throttle_from` update.
+	fn maybe_update_max(&mut self, new_max: Perbill) -> DispatchResult {
+		if let Some(old) = self.max.as_mut() {
+			if new_max > *old {
+				return Err(Error::<T>::MaxCommissionRestricted.into())
+			}
+			*old = new_max;
+		} else {
+			self.max = Some(new_max)
+		};
+		let updated_current = self
+			.current
+			.as_mut()
+			.map(|(x, _)| {
+				let u = *x > new_max;
+				*x = (*x).min(new_max);
+				u
+			})
+			.unwrap_or(false);
+
+		if updated_current {
+			self.register_update();
+		}
+		Ok(())
+	}
+
+	/// Set the pool's commission `change_rate`.
+	///
+	/// Once a change rate configuration has been set, only more restrictive values can be set
+	/// thereafter. These restrictions translate to increased `min_delay` values and decreased
+	/// `max_increase` values.
+	///
+	/// Update `throttle_from` to the current block upon setting change rate for the first time, so
+	/// throttling can be checked from this block.
+	fn maybe_update_change_rate(
+		&mut self,
+		change_rate: CommissionChangeRate<T::BlockNumber>,
+	) -> DispatchResult {
+		ensure!(!&self.less_restrictive(&change_rate), Error::<T>::CommissionChangeRateNotAllowed);
+
+		if self.change_rate.is_none() {
+			self.throttle_from = Some(<frame_system::Pallet<T>>::block_number());
```

Comment:
For the simplicity of code, it would be nice to call it something like `last_update` and always update it irrespective of whether throttle is set or not.

Reply:
The problem with `last_update` is that it is also set when there is no update to the commission - it is set when a change rate is initially set, and also if max commission pushes down current commission. So `last_update` is a little bit too vague IMO. It could also suggest to the untrained eye that `max` or `change_rate` could be subject to an update too.

`throttle_from` is the block to start throttling `min_delay` from, and is also used to calculate `intervals_passed`, which also effects `max_allowed_increase`.  I did ponder a while on this naming but ultimately think `throttle_from` is the most accurate representation of the value.